src/routes/event-match.tsx

```diff
@@ -91,28 +96,62 @@ const EventMatch = ({ eventKey, matchKey }: Props) => {
     [match],
   )
 
+  if (!isData(match)) {
+    return (
+      <Page
+        back={`/events/${eventKey}`}
+        name={
+          (m === null
+            ? matchKey + ' - '
+            : m.group + (m.num ? ' Match ' + m.num : '') + ' - ') +
+          (event ? event.name : eventKey)
+        }
+        class={matchStyle}
+      >
+        <Card class={matchStyle}>
+          <Icon icon={alert} />
+          <div
+            class={css`
+              font-size: 2rem;
+              text-align: center;
+            `}
+          >
+            {m === null
+              ? 'This Match Does Not Exist'
+              : m.group + (m.num ? ' Match ' + m.num : '') + ' Does Not Exist'}
+          </div>
+          <Button href={`/events/${eventKey}`}> Return to Event Page</Button>
+        </Card>
+      </Page>
+    )
+  }
+
   return (
     <Page
       back={`/events/${eventKey}`}
       name={
-        m.group +
-        (m.num ? ' Match ' + m.num : '') +
-        ' - ' +
+        (m === null
+          ? matchKey + ' - '
+          : m.group + (m.num ? ' Match ' + m.num : '') + ' - ') +
         (event ? event.name : eventKey)
       }
       class={matchStyle}
     >
       <Button href={`/events/${eventKey}/matches/${matchKey}/scout`}>
         Scout Match
       </Button>
-      {match ? <MatchCard match={match} eventKey={eventKey} /> : <Spinner />}
-      {match && matchHasBeenPlayed && (
+      {isData(match) ? (
```

Comment:
Here it is impossible for `isData(match)` to be false, because we already checked on line 99, and if it was false, we would have returned already.